### ISSUE TITLE
MissingConfigurationException after manually configured Amplify

### ISSUE THREAD

**Issue by dawidhyzy** [aws-amplify/amplify-ui-android#105]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui-android/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui-android/blob/main/CONTRIBUTING.md#reporting-bugsfeature-requests).

### Which UI component?

Authenticator

### Gradle script dependencies

```toml
// Put output below this line
amplifyframework-core = "com.amplifyframework:core-kotlin:2.14.5"
amplifyframework-ui-authenticator = "com.amplifyframework.ui:authenticator:1.0.2"
```

### Environment information

```
# Put output below this line
------------------------------------------------------------
Gradle 8.5
------------------------------------------------------------

Build time:   2023-11-29 14:08:57 UTC
Revision:     28aca86a7180baa17117e0e5ba01d8ea9feca598

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.9 (JetBrains s.r.o. 17.0.9+7-b1087.7)
OS:           Mac OS X 14.0 aarch64

```

### Describe the bug

We are building our configuration JSON using our backend response so I can't bundle `amplifyconfiguration.json` in the app.
Instead, I configure Amplify manually by:
```kotlin
Amplify.addPlugin(AWSCognitoAuthPlugin())
Amplify.configure(AmplifyConfiguration.fromJson(amplifyConfigurationJson), context)
```  
This works well with WebUI sign-in `Amplify.Auth.signInWithWebUI(activity)`. I want to migrate to amplify-ui-android but when I display the `Authenticator` composable I get the following: 
`MissingConfigurationException{message=Missing auth configuration, cause=null, recoverySuggestion=Make sure the Auth plugin is added and Amplify.configure is called. See https://docs.amplify.aws/lib/auth/getting-started/q/platform/android/ for details}`

Please note that manual configuration is performed **before** the `Authenticator` is shown.

### Code Snippet

```java
// Put your code below this line.

```

### Log output

```
// Put your logs below this line
Authenticator  E  MissingConfigurationException{message=Missing auth configuration, cause=null, recoverySuggestion=Make sure the Auth plugin is added and Amplify.configure is called. See https://docs.amplify.aws/lib/auth/getting-started/q/platform/android/ for details}
```

### amplifyconfiguration.json

```json
{
   "auth":{
      "plugins":{
         "awsCognitoAuthPlugin":{
            "IdentityManager":{
               "Default":{

               }
            },
            "CognitoUserPool":{
               "Default":{
                  "PoolId":"...sensitive...",
                  "AppClientId":"...sensitive...",
                  "Region":"eu-west-1"
               }
            },
            "Auth":{
               "Default":{
                  "authenticationFlowType":"USER_SRP_AUTH",
                  "OAuth":{
                     "WebDomain":"sensitive.amazoncognito.com",
                     "AppClientId":"...sensitive...",
                     "SignInRedirectURI":"sensitive:\/\/oauth",
                     "SignOutRedirectURI":"sensitive:\/\/logout\/",
                     "Scopes":[
                        "email",
                        "openid",
                        "profile"
                     ]
                  }
               }
            }
         }
      }
   }
}
```

**Comment by tjleing** [comment]
Hello @dawidhyzy, thanks for reaching out!  The authenticator component is supposed to use the passed-in configuration even if Amplify is configured with a runtime config, so this is not intended.  A member of the team will follow up when we discover more information about what's going on here.

**Comment by tylerjroach** [comment]
@dawidhyzy I believe you are likely running into the same issue as this ticket: https://github.com/aws-amplify/amplify-ui-android/issues/98.

The json is likely failing to be read properly, due to a missing or bad config. We have an improvement ticket to provide helpful messaging in this scenario, but in the meantime, I will try to provide a breakpoint to set within the authenticator library that should hopefully provide you with the cause.

**Comment by tjleing** [comment]
If you set a breakpoint in `AuthProvider.kt` `getConfiguration()`, specifically the catch block at the end of the function, you should be able to see what JSON error in particular is causing your configuration to not be read (assuming the `Amplify.configure()` is correct, which it looks like it is).

I was able to verify that a properly formatted `amplifyconfiguration.json` created by an `amplify init`/`amplify add auth` can be passed to `Amplify.configure(AmplifyConfiguration.fromJson(<JSONObject>))` and the authenticator will function correctly.

**Comment by mattcreaser** [comment]
I am updating Authenticator to emit a more useful error in this case, but for clarity to anyone reading this issue: the specific things missing from the posted configuration are: `signupAttributes`, `usernameAttributes`, `passwordProtectionSettings`, and `verificationMechanisms`.

These fields are used for the following:
- `signUpAttributes`: Populates the fields shown when signing up a new account
- `usernameAttributes`: Determines whether a username, phone number, or email address is used to sign in.
- `passwordProtectionSettings`: Sets the rules for which passwords will be accepted when signing up a new account.
- `verificationMechanisms`: Sets which account recovery mechanisms, if any, can be used to recover the account. The user may be prompted to verify their email, for example, after signing in if it is required for account recovery.

Here is an example configuration generated by Amplify that includes each of these fields (note I've left fields that aren't required as well):

```json
"Auth": {
    "Default": {
        "authenticationFlowType": "USER_SRP_AUTH",
        "socialProviders": [],
        "usernameAttributes": [],
        "signupAttributes": [
            "EMAIL"
        ],
        "passwordProtectionSettings": {
            "passwordPolicyMinLength": 8,
            "passwordPolicyCharacters": []
        },
        "mfaConfiguration": "OFF",
        "mfaTypes": [
            "SMS"
        ],
        "verificationMechanisms": [
            "EMAIL"
        ]
    }
}
```

**Comment by mattcreaser** [comment]
This improvement was released with Authenticator `1.1.0` 🎉
